server/pulp/server/lazy/url.py

```diff
@@ -0,0 +1,594 @@
+"""
+Classes that provide URL signing and validation.
+The approach is modeled after Amazon EC2 signing.
+
+Example:
+
+unsigned:
+
+http://redhat.com/content/test.rpm
+
+signed:
+
+http://redhat.com/content/test.rpm?policy=eyJyZXNvdXJjZSI6ICIvY29udGVudC90ZXN0LnJwbSIsICJleHBpcmF0a
+W9uIjogMTQ0MjkzMTc4OH0%3D;signature=PTgdjh0uq1lXDfpvAbYXxF_TO8BKHT4Eo6OYo6M45UL0jIokZ2P5s-tAclzAx-E
+Lv3x7kfjAADA8Cpsk9iHBeeq3pKTX6Ogv_R-BkORxJAAzeM7et9M1HkyGKwBl8BTYAA3POQT0Ips4Zb0LN5uDHFPwclS-W1jvKD
+095g__lhvLmiiVLw5biVM3WFvlUl0-hd4ljzbvB4qrXRGbNtw1rnbCIowM0h2_SsoS_WzetJWsiZNdMVySfUXKJ4m3-du89wWB7
+zDFYRlosVNTom5YiKprVkJEAo3cbTOMoZss_NS-vpzauuk-qXwqj05gI7fJVsVD915gGcY0xPUbpCbVTg%3D%3D
+"""
+
+from base64 import urlsafe_b64encode, urlsafe_b64decode
+from gettext import gettext as _
+from hashlib import sha256
+from time import time
+from urllib import quote, unquote
+from urlparse import ParseResult, urlparse, urlunparse
+
+from M2Crypto import RSA, BIO
+
+from pulp.server.compat import json
+
+
+class NotValid(Exception):
```

Comment:
Perhaps you could ditch this exception entirely, and base all of the ones below on ``ValueError`` ?